### PR TITLE
Add ability to modify conf file and directory permissions

### LINKED ISSUE #150: Cannot set syslog conf file and directory permissions
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet:  agent 6.4.3  (Puppet Enterprise 2019.1)
- Ruby:
- Distribution:  RHEL 7,8
- Module version:  4.0.0 (appears to be an issue in 5.0.0 as well)

## How to reproduce (e.g Puppet code you use)
include rsyslog::client

## What are you seeing
/etc/rsyslog.conf, /etc/rsyslog.d, /etc/rsyslog.d permissions cannot be set
We use 0600 on our files, but can't set that.

## What behaviour did you expect instead
Conf file Permissions exposed as a parameter

### PR DESCRIPTION
**Pull request by dhollinger**
#### Pull Request (PR) description
Add the ability to configure the file and directory permissions for generated configuration files and the confdir.

#### This Pull Request (PR) fixes the following issues

Fixes #150

### CODE REVIEW

**Comment by dhollinger** [comment]
@jdwhited could you review as well?

File: manifests/init.pp [hunk 1]
```diff
@@ -109,6 +113,8 @@
   Integer $ruleset_priority,
   Integer $filter_priority,
   String  $target_file,
+  String  $conf_permissions = '0644',
```
Comment (bastelfreak) [inline manifests/init.pp:116]: https://github.com/puppetlabs/puppetlabs-stdlib/blob/master/types/filemode.pp
```suggestion
  Stdlib::Filemode $conf_permissions = '0644',
```

File: manifests/init.pp [hunk 2]
```diff
@@ -109,6 +113,8 @@
...
+  String  $confdir_permissions = '0755',
```
Comment (bastelfreak) [inline manifests/init.pp:117]:
```suggestion
  Stdlib::Filemode $confdir_permissions = '0755',
```

File: manifests/init.pp [hunk 3]
```diff
@@ -79,36 +79,42 @@
...
+  Integer           $action_priority,
+  Integer           $input_priority,
+  Integer           $custom_priority,
+  Integer           $main_queue_priority,
+  Integer           $lookup_table_priority,
+  Integer           $parser_priority,
+  Integer           $ruleset_priority,
+  Integer           $filter_priority,
+  String            $target_file,
+  Stdlib::Filemode  $conf_permissions = '0644',
```
Comment (bastelfreak) [inline manifests/init.pp:116]: could be reduce by one whitespace

**Comment by jdwhited** [comment]
Thanks for working on this!  Can you allow perms for /etc/rsyslog.conf as well?  The other 2 look good.

**Comment by dhollinger** [comment]
> Thanks for working on this! Can you allow perms for /etc/rsyslog.conf as well? The other 2 look good.

Added

**Pull request merged**